### PR TITLE
ipn/ipnlocal: do controlclient.Shutdown in a different goroutine

### PR DESCRIPTION
**Pull request by maisem**
We do not need to wait for it to complete. And we might have to call Shutdown from callback from the controlclient which might already be holding a lock that Shutdown requires.

Updates #713

Signed-off-by: Maisem Ali <maisem@tailscale.com>

### CODE REVIEW

File: ipn/ipnlocal/local.go [hunk 1]
```diff
@@ -4053,6 +4058,7 @@ func (b *LocalBackend) resetForProfileChangeLockedOnEntry() error {
 	b.setNetMapLocked(nil) // Reset netmap.
 	// Reset the NetworkMap in the engine
 	b.e.SetNetworkMap(new(netmap.NetworkMap))
+	b.resetControlClientLocked()
```
Comment (mihaip) [inline ipn/ipnlocal/local.go:4061]: Is this necessary? We end up calling `resetControlClientLocked` in `Start` either way.

File: ipn/ipnlocal/local.go [hunk 2]
```diff
@@ -3238,6 +3236,17 @@ func (b *LocalBackend) requestEngineStatusAndWait() {
 	b.statusLock.Unlock()
 }
 
+func (b *LocalBackend) resetControlClientLocked() {
+	if b.cc == nil {
+		return
+	}
+	// Call Shutdown in a goroutine, as it may require
+	// locks that we're holding. We do not need to wait
```
Comment (mihaip) [inline ipn/ipnlocal/local.go:3244]: Is it locks that we're holding? Based on the stack that I observed, the deadlock was contained with the control client itself.

**Review by mihaip — APPROVED** [review]
This is a lot simpler than #6324, so it seems preferable.

File: ipn/ipnlocal/local.go (hunk 2 above)
Comment (maisem) [reply ipn/ipnlocal/local.go]: I thought it was a lock but I could be wrong, reworded regardless.

File: ipn/ipnlocal/local.go (hunk 1 above)
Comment (maisem) [reply ipn/ipnlocal/local.go]: Good point. removed.

**Comment by maisem** [comment]
@mihaip @bradfitz I had to modify TestStateMachine as it would never recreate the mockControl client and expected "Shutdown" and "New" events in order. However as we are now calling Shutdown in a goroutine, there was no way to guarantee the ordering, so instead I made the test also recreate the client when it was Shutdown. To ensure that shutdown was still being invoked I keep track of the previous client and make sure it is shutdown.

I also removed all the unnecessary "pause"/"unpause" and now only do them when it toggles state. This allowed me to remove all the places where we had `"unpause", "unpause", "unpause"`

**Review by bradfitz — APPROVED** [review]

**Pull request merged**